packages/fxa-settings/src/components/Settings/ConnectedServices/index.tsx

```diff
@@ -106,7 +106,8 @@ export const ConnectedServices = forwardRef<HTMLDivElement>((_, ref) => {
         // disconnect all clients/sessions with this name since only unique names
         // are displayed to the user. This is batched into one network request
         // via BatchHttpLink
-        const clientsWithMatchingName = groupedByName[client.name || 'unknown'];
+        const groupByKey = client.name || '';
+        const clientsWithMatchingName = groupedByName[groupByKey];
         const hasMultipleSessions = clientsWithMatchingName.length > 1;
         if (hasMultipleSessions) {
```

Comment:
I know this isn't part of the issue, whats up with the flag though? Seems like simply looping over the collection would be fine :confused:

Reply:
It probably would be fine to loop over the clients, maybe a little slower since each disconnect needs to resolve before disconecting the next client?